client/lib/directly/index.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import config from 'config';
+
+/**
+ * Internal dependencies
+ */
+import { loadScript } from 'lib/load-script';
+
+const DEFAULT_RTM_WIDGET_OPTIONS = {
+	id: config( 'directly_rtm_widget_id' ),
+	displayAskQuestion: false
+};
+const DIRECTLY_RTM_SCRIPT_URL = 'http://widgets.wp.com/directly/embed.js';
+const DIRECTLY_ASSETS_BASE_URL = 'https://www.directly.com';
+let initializationPromise;
+
+function insertDirectlyDOM() {
+	// Directly gathers the base URL for its assets by inspecting the src attribute
+	// of a DOM element with id="directlyRTMScript". Since we're not using the standard
+	// integration, we need to fake it by adding a dummy DOM object with those attributes.
+	const d = document.createElement( 'div' );
+	d.id = 'directlyRTMScript';
+	d.src = DIRECTLY_ASSETS_BASE_URL;
+	document.body.appendChild( d );
+}
+
+function executeDirectlyCommand( ...args ) {
+	// Havoc is wreaked if you try to execute Directly commands before initialization.
+	// If initialization hasn't started, ignore the command.
+	if ( typeof initializationPromise === 'undefined' ) {
+		return;
+	}
+	// If initialization has started, add the command to the promise chain to execute
+	// as soon as initialization has finished.
+	return initializationPromise.then( () => {
+		window.DirectlyRTM( ...args );
+	} );
+}
+
+export function initialize( directlyConfig = {} ) {
+	if ( typeof initializationPromise !== 'undefined' ) {
+		return initializationPromise;
+	}
+	// Set up the global DirectlyRTM function, required for the Directly library
+	window.DirectlyRTM = window.DirectlyRTM || function() {
+		( window.DirectlyRTM.cq = window.DirectlyRTM.cq || [] ).push( arguments );
+	};
+	// Before loading the script, we need to enqueue the config details...
+	window.DirectlyRTM( 'config', { ...DEFAULT_RTM_WIDGET_OPTIONS, ...directlyConfig } );
+	// ... and prepare the DOM for Directly
+	insertDirectlyDOM();
+
+	initializationPromise = new Promise( ( resolve, reject ) => {
+		loadScript( DIRECTLY_RTM_SCRIPT_URL, function( error ) {
+			if ( error ) {
+				reject( error );
+			} else {
+				resolve();
+			}
+		} );
+	} );
+
+	return initializationPromise;
```

Comment:
We weren't initializing the way you suggest for a couple reasons:
1. As soon as `lib/directly` is `import`ed you're making an HTTP request that you may not need (from the `loadScript`).
2. That requested script has side-effects on parsing — the widget will appear immediately for users who recently interacted with it, which isn't desirable in all circumstances.
3. This doesn't allow app code to pass in specific configuration options (configuration needs to happen before `loadScript`).

The more I think about it though, since there can only be one `config` per pageload, it probably doesn't make sense for components to be able to specify their own configuration... Say ComponentA and ComponentB specify different config options — it's not transparent that one will win and the other will be ignored. So removing custom config is probably OK and solves that problem.

And in that case, yeah it seems to make a lot of sense for each command to just start initialization if it hasn't already happened. We may want to still keep an `initialize` command present for pre-fetching the library.